src/App.tsx

```diff
@@ -1,56 +1,62 @@
-import React from 'react';
-import './App.scss';
+import React, { useMemo, useState } from 'react';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+import { DropdownList } from './components/DropdownList';
+import { InputField } from './components/InputField';
 
-  return (
-    <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+import { Person } from './types/Person';
 
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
+import './App.scss';
 
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
+export const App: React.FC = () => {
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
 
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
+  const filteredPeoples = useMemo(() => {
+    const filtered = peopleFromServer.filter(person => {
+      const { name } = person;
 
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
+      return name.toLowerCase().includes(query.toLowerCase());
+    });
 
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
+    if (filtered.length === 0) {
+      return null;
```

Comment:
Returning `null` for `filteredPeoples` when there are no matches might be a problem. It would be better to return an empty array to prevent potential errors in the `DropdownList` component where you are likely iterating over `peoples`.